sljit_src/sljitExecAllocator.c

```diff
@@ -142,25 +137,27 @@ static SLJIT_INLINE int get_map_jit_flag()
 	return map_jit_flag;
 }
 #endif /* MAP_ANON */
-#else /* !x86_macOS */
-#define SLJIT_MAP_JIT	(MAP_JIT)
-#endif /* TARGET_OS_OSX && SLJIT_CONFIG_X86 */
-#else /* !MAP_JIT */
-#define SLJIT_MAP_JIT	(0)
-#endif /* MAP_JIT */
-
-#if defined(MAC_OS_X_VERSION_MAX_ALLOWED) \
-	&& MAC_OS_X_VERSION_MAX_ALLOWED >= 110000
+#else /* !SLJIT_CONFIG_X86 */
+#if !(defined SLJIT_CONFIG_ARM && SLJIT_CONFIG_ARM)
+#error Unsupported architecture
+#endif /* SLJIT_CONFIG_ARM */
```

Comment:
technically not needed and could be made more strict by checking for ARM_64, but will hopefully help when apple does its next CPU migration